src/engine/runtime.js

```diff
@@ -830,18 +825,24 @@ class Runtime extends EventEmitter {
     }
 
     /**
-     *  Read extension information, convert menus, blocks and custom field types
+     * Read extension information, convert menus, blocks and custom field types
      * and store the results in the provided category object.
      * @param {CategoryInfo} categoryInfo - the category to be filled
      * @param {ExtensionMetadata} extensionInfo - the extension metadata to read
      * @private
      */
     _fillExtensionCategory (categoryInfo, extensionInfo) {
+        categoryInfo.blocks = [];
+        categoryInfo.customFieldTypes = {};
```

Comment:
Any particular reason for moving this out of `_registerExtensionPrimitives` and moving it here instead?

Reply:
`_fillExtensionCategory` is also called by `_refreshExtensionPrimitives` so we need to clear these out to prevent duplicating things on refresh. Previously `_refreshExtensionPrimitives` was clearing blocks and menus but not custom fields, so it was an easy problem to miss. By putting these here it's harder to forget to clear a new `categoryInfo` field in `_registerExtensionPrimitives` before filling it.